### PR TITLE
Fix `Style/RedundantArgument` cop failure while inspecting string literal with invalid encoding

### PR DESCRIPTION
**Pull request by viralpraxis**
We have (roughly) this code in our test sutie:

```ruby
s.chomp("\x82")
```

it fails with this backtrace:

```
An error occurred while Style/RedundantArgument cop was inspecting test.rb:1:0.
invalid byte sequence in UTF-8
lib/rubocop/cop/style/redundant_argument.rb:121:in `sub'
lib/rubocop/cop/style/redundant_argument.rb:121:in `exclude_cntrl_character?'
lib/rubocop/cop/style/redundant_argument.rb:111:in `argument_matched?'
lib/rubocop/cop/style/redundant_argument.rb:90:in `redundant_argument?'
lib/rubocop/cop/style/redundant_argument.rb:67:in `on_send'
```

Seems like we have to check if provided value's encoding is valid before performing string operations.

-----------------

Before submitting the PR make sure the following are checked:

* [X] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [X] Wrote [good commit messages][1].
* [X] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [X] Feature branch is up-to-date with `master` (if not - rebase it).
* [X] Squashed related commits together.
* [X] Added tests.
* [X] Ran `bundle exec rake default`. It executes all tests and runs RuboCop on its own code.
* [X] Added an entry (file) to the [changelog folder](https://github.com/rubocop/rubocop/blob/master/changelog/) named `{change_type}_{change_description}.md` if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: spec/rubocop/cop/style/word_array_spec.rb [hunk 1]
```diff
@@ -559,7 +559,7 @@
       RUBY
     end
 
-    it "doesn't fail with `encoding: binary" do
+    it "doesn't fail with `encoding: binary`" do
```
Comment (koic) [inline spec/rubocop/cop/style/word_array_spec.rb:562]: This change is unrelated to this PR, right? It might be better to open it as a separate PR.

Comment (viralpraxis) [reply spec/rubocop/cop/style/word_array_spec.rb]: Sure, that was just an occasional style fix.
